packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -1085,6 +1085,8 @@ class Auth {
             const errorData = await response.json()
             throw new Error(errorData.message || 'API validation failed')
         }
+        // Await 1s before calling authorizeIDP again
+        await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
🤔 The side effect of having this 1 sec delay is well, `authorizeIDP` call now takes 1 sec long. This is not good for performance in general. So it feels like a workaround, rather than a proper fix.

What is the reason for requesting `url` twice in a row? I see there's a fetch on line 1079 and then we ask the browser to navigate to the same url on line 1091.

Is it possible to change the logic to request that url only once?

Reply:
@vmarta The reason I added an initial fetch is so we could handle errors in the storefront before we assign the url in the window... Without doing this initial fetch, we get an error on the redirect page with just the error on the page (no UI)...

Reply:
Related to this, I've made a PR to show immediate feedback upon clicking the social login button: https://github.com/SalesforceCommerceCloud/pwa-kit/pull/2242

Can you please review it when you have time? Thanks.